src/category_theory/equivalence.lean

```diff
@@ -7,6 +7,40 @@ import category_theory.fully_faithful
 import category_theory.whiskering
 import tactic.slice
 
+/-!
+# Equivalence of categories
+
+An equivalence of categories `C` and `D` is a pair of functors `F : C ⥤ D` and `G : D ⥤ C` such
+that `η : 𝟭 C ≅ F ⋙ G` and `ε : G ⋙ F ≅ 𝟭 D`. In many situations, equivalences are a better
+notion of "sameness" of categories than the stricter isomorphims of categories.
+
+Our definition of equivalence includes an additional axiom, and the resulting notion of equivalence
+is called a "(half-)adjoint equivalence". We show that, given an equivalence in the sense of the
```

Comment:
Most practicing mathematicians know what an equivalence is, and what an adjunction is. But I guess that many don't realise that there are equivalences that are not adjunctions. (I only realised this when I formalised them :grinning:)

Would you mind explaining in a bit more detail what the subtleties are? (Maybe part of this should even go in the docstring of equivalence, or it could have a pointer to this module docstring.)

For example, I don't know what the difference is between a `half-adjoint equivalence` and an `adjoin equivalence`. And which one does mathlib have?

Reply:
I tried to expand the explanation a bit. Is it a bit clearer now?